### PR TITLE
Merge Slicer plugin to master

### PR DESCRIPTION
**Pull request by SachidanandAlle**
1) Fix py-client APIs for Clara 2.0 release (with backward compatibility)
2) Add Slicer Plugin

@YuanTingHsieh

### CODE REVIEW

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import logging
+
+
+class SegmentEditorNvidiaAIAA(ScriptedLoadableModule):
+    """Uses ScriptedLoadableModule base class, available at:
+    https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def __init__(self, parent):
+        import string
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py]: what is this for?

File: py-client/client_api.py [hunk 2]
```diff
@@ -88,30 +90,48 @@ def model_list(self, label=None):
...
+            points = json.loads(form.get('points'))
+            params = {'points': (json.loads(points) if isinstance(points, str) else points)}
+        else:
+            params = json.loads(params) if isinstance(params, str) else params
+
         AIAAUtils.save_result(files, image_out)
-        return form
+
+        if self.doc_id is None:
+            self.doc_id = form.get('doc') if isinstance(form, dict) else None
```
Comment (YuanTingHsieh) [inline py-client/client_api.py]: is this the check we want?
I thought after Line 120, form should be dict?
and is form guaranteed to have 'doc'?

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        segmentEditorWidget.setMRMLScene(slicer.mrmlScene)
+        segmentEditorNode = slicer.vtkMRMLSegmentEditorNode()
+        slicer.mrmlScene.AddNode(segmentEditorNode)
+        segmentEditorWidget.setMRMLSegmentEditorNode(segmentEditorNode)
+        segmentEditorWidget.setSegmentationNode(segmentationNode)
+        segmentEditorWidget.setMasterVolumeNode(masterVolumeNode)
+
+        ##################################
+        self.delayDisplay("Run segmentation")
+        segmentEditorWidget.setActiveEffectByName("NvidiaAIAA")
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py]: where can we find this effect name

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+from SegmentEditorEffects import *
+
+from AIAAClient import AIAAClient
+
+
+class SegmentEditorEffect(AbstractScriptedSegmentEditorEffect):
+    """This effect uses Watershed algorithm to partition the input volume"""
+
+    def __init__(self, scriptedEffect):
+        scriptedEffect.name = 'Nvidia AIAA'
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: should this name matched the name we have upthere for testing?

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+            qt.QApplication.setOverrideCursor(qt.Qt.WaitCursor)
+            logic = AIAALogic(self.serverUrl.text,
+                              progress_callback=lambda progressPercentage,
+                                                       progressBar=progressBar: SegmentEditorEffect.report_progress(
+                                  progressBar, progressPercentage))
+
+            inputVolume = self.scriptedEffect.parameterSetNode().GetMasterVolumeNode()
+            extreme_points, result_file = logic.segmentation(model, inputVolume)
+            if self.updateSegmentationMask(extreme_points, result_file, modelInfo):
+                result = 'SUCCESS'
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: is this variable `result` necessary?
or just a literal is enough

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+                              progress_callback=lambda progressPercentage,
+                                                       progressBar=progressBar: SegmentEditorEffect.report_progress(
+                                  progressBar, progressPercentage))
+
+            inputVolume = self.scriptedEffect.parameterSetNode().GetMasterVolumeNode()
+            modelInfo = self.models.get(model)
+            pointSet = self.getFiducialPointsXYZ()
+
+            result_file = logic.dextr3d(model, pointSet, inputVolume, modelInfo)
+            result = 'FAILED'
```
Comment (YuanTingHsieh) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: is this variable `result` necessary?
or just a literal is enough
cuz we are not using it if something actually failed

File: py-client/client_api.py (hunk 2 above)
Comment (SachidanandAlle) [reply py-client/client_api.py]: Fixed

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py (hunk 3 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAA.py]: It's part of slicer plugin template

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 4 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: It's part of slicer plugin template

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 5 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: It's just for the logging/message

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py (hunk 6 above)
Comment (SachidanandAlle) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: It's just for the logging/message

**Pull request merged**